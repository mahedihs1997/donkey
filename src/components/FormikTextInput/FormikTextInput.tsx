import React from "react";
import { useField } from "formik";
import { TextField } from "@material-ui/core";

const FormikTextInput = ({ label, ...props }) => {
  const [field, meta] = useField(props);
  return (
    <>
      <TextField {...field} {...props} placeholder={label} />
      <p>{meta.touched && meta.error && meta.error}</p>
    </>
  );
};

export default FormikTextInput;
