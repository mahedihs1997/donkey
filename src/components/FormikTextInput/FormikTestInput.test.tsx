import { Formik } from 'formik';
import React from 'react';
import TestRenderer from 'react-test-renderer';

import FormikTextInput from '.';

describe('< FormikTextInput />', () => {
  it('should render without throwing an error and match snapshot', () => {
    const testRenderer = TestRenderer.create(
      <Formik initialValues={{ test: '' }} onSubmit={jest.fn()}>
        <FormikTextInput name="Test" />
      </Formik>
    );
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
