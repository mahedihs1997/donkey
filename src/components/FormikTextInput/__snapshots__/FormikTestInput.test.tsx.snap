// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< FormikTextInput /> should render without throwing an error and match snapshot 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root makeStyles-input-1 MuiFormControl-fullWidth"
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input"
      disabled={false}
      name="Test"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
  </div>
  <p
    className="MuiFormHelperText-root"
  >
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "&#8203;",
        }
      }
    />
  </p>
</div>
`;
